api/v1beta1/oscmachine_validation.go

```diff
@@ -119,16 +119,19 @@ func ValidateDeviceName(path *field.Path, deviceName string) *field.Error {
 	}
 }
 
-var isValidateVmType = regexp.MustCompile(`^tinav([3-9]|[1-9][0-9]).c[1-9][0-9]*r[1-9][0-9]*p[1-3]$`).MatchString
+var isValidTinaVmType = regexp.MustCompile(`^tinav([3-9]|[1-9][0-9]).c[1-9][0-9]*r[1-9][0-9]*p[1-3]$`).MatchString
+var isValidInferenceVmType = regexp.MustCompile(`^inference7-(?:l40\.(?:medium|large)|h100\.large|h200\.4xlargeA)$`).MatchString
 
 // ValidateVmType checks that vmType is a valid vmType
 func ValidateVmType(path *field.Path, vmType string) *field.Error {
 	switch {
 	case vmType == "":
 		return field.Required(path, "vmType is required")
-	case isValidateVmType(vmType):
+	case isValidTinaVmType(vmType):
+		return nil
+	case isValidInferenceVmType(vmType):
 		return nil
 	default:
-		return field.Invalid(path, vmType, "vmType must use the tinavX.cXrXpX format")
+		return field.Invalid(path, vmType, "vmType must use either the tinavX.cXrXpX or inferenceX.gpuX.sizeX format")
```

Comment:
All Xs are numbers, gpuX and sizeX do not seem to be the right pattern
`inferenceX.gpuX.sizeX ` => `inferenceX-gpu.size` or `inferenceX-{gpu}.{size}` ?